tests/integration/test_dbapi_integration.py

```diff
@@ -1782,6 +1783,43 @@ def test_rowcount_insert(trino_connection):
         assert cur.rowcount == 1
 
 
+@pytest.mark.parametrize(
+    "legacy_prepared_statements",
+    [
+        True,
+        pytest.param(False, marks=pytest.mark.skipif(
+            trino_version() <= '417',
+            reason="EXECUTE IMMEDIATE was introduced in version 418")),
+        None
+    ]
+)
+def test_prepared_statement_capability_autodetection(legacy_prepared_statements, run_trino):
+    # start with an empty cache
+    trino.dbapi.must_use_legacy_prepared_statements = TimeBoundLRUCache(1024, 3600)
+    user_name = f"user_{t.monotonic_ns()}"
+
+    _, host, port = run_trino
+    connection = trino.dbapi.Connection(
+        host=host,
+        port=port,
+        user=user_name,
+        legacy_prepared_statements=legacy_prepared_statements,
+    )
+    cur = connection.cursor()
+    cur.execute("SELECT ?", [1])
+    cur.fetchall()
+    cur.execute("SELECT ?", [42])
+    cur.fetchall()
```

Comment:
Is 2 queries on single connection testing something? We do test with 2 cursors just after this so we do test cache working properly just with that.

```suggestion
    cur.execute("SELECT ?", [42])
    cur.fetchall()
```

Reply:
If i missed something which the two queries are testing then please add a comment since it's not obvious

Reply:
No, I started with the first 2 queries and then I realized I should test different connections and didn't go back to remove one of the first 2.